components/styled/table/src/index.tsx

```diff
@@ -1,20 +1,24 @@
-import { FC } from 'react'
+import GenericTable, { TableData } from './atoms/table'
 
-import { Questionnaire, QuestionnaireResponse, SummaryTableViewType } from '@ltht-react/types'
-import QuestionnaireTable from './molecules/questionnaire-table'
-
-const Table: FC<IProps> = ({ definition, records, orientation = 'VERTICAL' }) => {
-  if (!definition.item || definition.item.length === 0) {
+export default function Table<TColumn, TRow>({ columnData, rowData, mapToTableData }: IProps<TColumn, TRow>) {
+  if (
+    !columnData ||
+    !rowData ||
+    (Array.isArray(columnData) && columnData?.length === 0) ||
```

Comment:
Equally I'd avoid type checking at runtime - the main benefit of Typescript is that we don't have to because we have the type checking at compile time